Add `Opens in new Tab` control into Link Preview

getdave opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

Adds the ability to toggle the `Opens in new tab` link setting from the preview step of the Link UI.

~Also keeps the Link UI open when toggling the checkbox.~ This has been deferred until a unrelated bug can be resolved.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

User feedback is that toggling this control is extremely common and should be made easier. Exposing it in the preview will enable this whilst not cluttering the initial link creation step.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

Add render slot into the bottom of the Link Preview component and expose the `LinkSettings` control there exposing only the `opensInNewTab` setting. 

When toggled the `onChange` of the component will be called with only that value present.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

- New Post
- Add text
- Create link on text
- When submitted see the "preview" is immediately displayed
- See the `Opens in new tab` checkbox in the preview.
- Toggle the checkbox. See the preview remains open/active.
- Click away from link.
- Click back to link. Check that Opens in new tab settings was retained.
- Click "Edit" on link. Check Opens in new tab setting matches in edit mode.
- Toggle the setting in edit mode and `Save`.
- Check that the setting shown on preview is correct.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->


https://github.com/WordPress/gutenberg/assets/444434/765e08db-eb4c-40b6-994d-290713d45f93

getdave committed:
Add opensInNewTab to Link Preview step

getdave committed:
Remove settings state tracking from inline link handling

getdave committed:
Ensure only opensInNewTab can be exposed in preview

getdave commented on packages/format-library/src/link/inline.js:
```diff
@@ -82,32 +71,18 @@ function InlineLinkUI( {
 	}
 
 	function onChangeLink( nextValue ) {
```
These changes have broken the ability to add new links so we'll need to fix that 😓

getdave replied:
✅ Fixed

getdave committed:
Reinstate closing of Link UI

getdave commented:
Ideally I would prefer the preview not to close when the checkbox is checked but for the value to still be applied. I've looked and I cannot find what is triggering that behaviour.

getdave committed:
Do not make formats inactive when creating link

ricjcs commented:
Excellent! That's the best solution.

It basically stays as it was before and additionally has advanced options.

Now yes, that solves the problem.

Thank you very much for listening to users and for the work!

richtabor commented:
There's potential here. The popover moving to the upper left corner is odd; not sure what's going on there. The checkbox also does not save if you edit it within that state. But if you click on the link again, it does. 

https://github.com/WordPress/gutenberg/assets/1813435/56986d68-78d1-4089-9140-86a9eb1855a8

richtabor commented:
And when `is-preview`, `.block-editor-link-control__setting` padding should be `20px 8px 8px 0`. 

<img width="350" alt="CleanShot 2023-08-11 at 10 23 02" src="https://github.com/WordPress/gutenberg/assets/1813435/f7ae4e72-7edc-4d47-b84c-17a122924246">

getdave commented:
That is very odd. I hadn't noticed that on my small screen. Will take a look.

richtabor commented:
> That is very odd. I hadn't noticed that on my small screen. Will take a look.

It's actually doing it in your video—moving to the top left. But it's not as noticeable at those dimensions.

elgato91 commented:
Compared to the previous interface, the current one is pejorative. It is no longer clear or simple and introduces additional steps, actions, and waiting time to perform required tasks. We should thoroughly investigate the reasons behind this change. What are the benefits it brings?

Furthermore, we should also consider that while we frequently discuss the concept of "open in a new window" it is equally important to implement controls for the "no follow" attribute. This attribute holds significant importance in terms of SEO compliance, aligning with recent Google guidelines pertaining to sponsored links, affiliations, and similar aspects.

Additionally, the checkboxes appear to represent a design evolution from the previous toggle switches. Moreover, what advantages does the anchor text field in the advanced section bring?

getdave commented:
Thanks for your questions. Usually we'd keep the feedback related to the proposal in the PR but as this relates to wider concerns I'll address some of your queries to enable you to better review this specific solution in context.

> Compared to the previous interface, the current one is pejorative.

Are you referring to the UI proposed in this PR? 

> It is no longer clear or simple... 

Can you go deeper here? It _sounds_ a little subjective. 

To me the UI itself appears to be very simple when first creating a link (witness lack of other controls). It could therefore reasonably be argued that showing additional controls here would make the interface more complex. What specifically is not clear/simple in the steps shown in _this PR_?

> ...and introduces additional steps, actions, and waiting time to perform required tasks.

There is a valid discussion to be had here. What would your proposal be? I assume it would be to show the `Opens in new tab` control _immediately_ at link creation step once the user has entered a value into the URL input?

> ...it is equally important to implement controls for the "no follow" attribute

The work we are doing is geared towards allowing more controls such as this. However we cannot continue to expand the number of controls shown on the first step indefinitely. This is precisely the background to why we moved the `Opens in new tab`.

Consider that other contributors may suggest other controls which - from their perspective - are equally important to _them_ and before we know it the UI would be incredibly complex. Therefore as contributors we have to make difficult decisions to deprioritise the majority of controls to avoid overwhelming the user.

Our current thinking is that whilst `Opens in new tab` _could_ be a special case (considering how long it's been in WordPress), all other controls should be moved into the `Advanced` area. Whilst it might be important for one set of users, for another set this control would simply add additional complexity.

With the auto-saving of the expanded state of this control already merged, this seems like a reasonable compromise solution.

Implementing `No follow` will be something we may do once the UI has reached a stable state.

> Additionally, the checkboxes appear to represent a design evolution from the previous toggle switches

The checkboxes are an evolution and one that was undertaken in a deliberate manner based on user feedback. 

The reason checkboxes are used is because they imply a need to _submit_ a change. We had numerous reports of confusion with the previous toggle-based interface where changes to settings would immediately cause the value to be submitted in unexpected ways. This is why we moved to a checkbox and also requiring the user to submit their changes. Obviously if we choose to pursue this PR then that distinction is less clear cut but I think it still works. Do you? If not why.


> what advantages does the anchor text field in the advanced section bring?

I didn't follow this. If it's not directly related to the change in this particular PR I would recommend [raising a new Issue](https://github.com/WordPress/gutenberg/issues/new/choose) documenting any problems you've experienced. Many thanks.

getdave commented:
Re: incorrect placement of popover. This seems to be because on first render the popover's `anchor` is not the link `<a>` but rather then entire `<p>` paragraph block. When you click _again_ the popover's `anchor` is updated to render to the `<a>` tag.

See graphic below which shows the console logging of first and then second clicks:

<img width="1120" alt="Screen Shot 2023-08-14 at 12 35 06" src="https://github.com/WordPress/gutenberg/assets/444434/0b8ce266-0313-4676-9153-956b24e03959">

Interesting when you first create a link the popover knows how to position itself correctly relative to the selection so now looking at how we loose that information once the link is created.

elgato91 commented:
Thanks, Dave. Your answers are greatly appreciated. I am very sorry that I discovered the new Link UI issue only after being impacted by it because I would have gladly contributed since its inception. Obviously, the choice between switches or checkboxes works well in any case, and I think you are right to favor what gives greater stability.

Regarding the design and concept of the Link UI, I believe there is some reasoning to be done on the matter — reasoning that is not subjective but based on factual issues. The before-and-after comparisons, in my view, are between the old UI and the new UI. The old UI was well-designed because it did its job: inserting a link, possibly searching for an existing page and linking to it, and marking the link with the two most important tags: opening in new tab and adding the nofollow tag.

While "open in new tab" is obviously used and abused even by amateur users, the latter is extremely important for anyone who understands the rudiments of SEO. Publishers, bloggers, marketers, and so on need (and it is a fundamental requirement of the Google Guidelines) to be able to add the nofollow tag very frequently — whenever there is a commercial relationship (affiliation or other) and whenever they are linking to a resource they don't fully trust. In the professional context, most links are often "nofollow", and it is also the most time-sensitive context since any additional action (click) requires more effort (and time) from the content creator.

This is not a subjective matter; a simple web search or reading the Google guidelines would reveal it. The old UI also allowed the sponsored tag to be included, strictly following the new Google guidelines. However, it currently appears that nofollow and sponsored can work equally well for Google.

> To me the UI itself appears to be very simple when first creating a link (witness lack of other controls). It could therefore reasonably be argued that showing additional controls here would make the interface more complex. What specifically is not clear/simple in the steps shown in this PR?

Coming to that, I think that's the point. A minimal interface does not equate to an easier to use interface. You're just moving vital controls elsewhere, sweeping the dust under the carpet. This, while aesthetically it looks like a simplification, has actually significantly increased the complexity of the interface.

> Consider that other contributors may suggest other controls which - from their perspective - are equally important to them and before we know it the UI would be incredibly complex. Therefore as contributors we have to make difficult decisions to deprioritise the majority of controls to avoid overwhelming the user.

There's no making up to say what's the best way to handle this, the old UI did it brilliantly! The reasoning made for the design of the old UI is still valid: inserting links and immediately choosing to mark them with "open in a new tab" and "no follow" are the essential controls. It's something that has always existed in WP.

getdave commented:
@elgato91 Thanks for your thoughts. To keep this PR's comments on track I'm going to recommend you/we re-post your comment on [the Issue which I found which tracks the "No follow" requirement](https://github.com/WordPress/gutenberg/issues/13542#issuecomment-1677256056).



In terms of UI design I think this discussion can continue on https://github.com/WordPress/gutenberg/issues/50891.

Thanks again for your feedback and contributing to the project.

getdave commented:
My new thinking re: [this bug](https://github.com/WordPress/gutenberg/pull/53566#issuecomment-1677160323) is that we loose the "selection" in the paragraph block when a link is submitted. That means the virtual anchor for the popover doesn't know where to "anchor" itself to. 

This is in contrast to:

- original link creation - you select the text and then adding a link causes the popover to be anchored to that selection range
- the link preview - this is anchored to the `<a>` tag once that is available.

That's why we end up with a popover in the top left corner. I don't yet know how to fix that. Ideally when the link is submitted the `<a>` should already be in the document but it appears it isn't added until it's too late...

### Update: 15th August

More deep diving here has revealed that the `useLayoutEffect` of `useAnchor` which is pretty much what determines where the popover is anchored is called once when the link _creation_ popover UI is shown but is _not_ called again when the link is _submitted_.

https://github.com/WordPress/gutenberg/blob/a7327a11f79135300cd69f40a797c866959a62ad/packages/rich-text/src/component/use-anchor.js#L132

This means that the calculation to find the new anchor point based on the new `<a>` (the link format) within the rich text does not happen. This means that because we no longer have a text selection in the richtext, the popover is positioned relative to the entire rich text paragraph. 

One "fix" I found is to call `callback()` immediately within the `attach` function:

https://github.com/WordPress/gutenberg/blob/a7327a11f79135300cd69f40a797c866959a62ad/packages/rich-text/src/component/use-anchor.js#L143-L145

The strange thing is why `callback()` doesn't get called anyway by the event listener which is added by the `attach` function:

https://github.com/WordPress/gutenberg/blob/a7327a11f79135300cd69f40a797c866959a62ad/packages/rich-text/src/component/use-anchor.js#L144

Given that is based on the `selectionchange` event, you would assume that when the focus moves from the richtext to the popover the selection change event would trigger. 

How this doesn't happen because the event listener is deregistered by the `detach` event which is called as soon as the richtext looses focus and moves to the `LinkControl`.

https://github.com/WordPress/gutenberg/blob/a7327a11f79135300cd69f40a797c866959a62ad/packages/rich-text/src/component/use-anchor.js#L147-L149

richtabor commented:
Should this work with #53642 applied? I'm trying it combined here: 

https://github.com/WordPress/gutenberg/assets/1813435/56f849cc-8c93-48c7-afa6-ad4dbddd12c2

EDIT: 

> That's why we end up with a popover in the top left corner. I don't yet know how to fix that. Ideally when the link is submitted the <a> should already be in the document but it appears it isn't added until it's too late...

Ah, gotcha. I think this is improvement, even with the popover closing like previously. It reduces another action to get to the "Add to new tab" control. Would appreciate a look from @WordPress/gutenberg-design as well though.

getdave commented:
> Ah, gotcha. I think this is improvement, even with the popover closing like previously

Suggestion - let's roll back the change that leaves the popover open and ship the improvement to add the `Opens in new tab` to the preview.

Then I'll follow up to try and work out the weirdness with positioning that I've described above.

getdave committed:
Reinstate auto-hide of UI after link submission as temp patch

Bug identified. Decided to ship what we had.

See https://github.com/WordPress/gutenberg/pull/53566#issuecomment-1678638946

getdave commented:
This is the new experience with the auto-closing reinstated. Clearly this is not as good but it's a stepping stone whilst we figure out this positioning problem.

https://github.com/WordPress/gutenberg/assets/444434/f9f695e8-ec25-48ef-9fda-d77b8987fa7d

TapiwaZvaks commented:
@getdave thanks your last screengrab does seem to be an improvement. Am speaking from the point of view of a heavy user, so please, bear with me. Am not sure I follow what magic you guys do behind the scenes, but it's much appreciated. Was just testing the solution that was implemented in the Gutenberg plugin (the one that leaves the drawer open). Again, it's something of an improvement. 

Still, I am in total agreement with @elgato91. It appears you guys are concerned about aesthetics at the expense of functionality. As an example, does anyone really need that link preview? It's not necessary (in my view). Also, I don't know what the thinking was behind this entire trajectory. 

What is it about opening a link in a new tab that got us here? To say that new tabs are bad is rather counterintuitive in SEO terms. I don't want someone to immediately abandon my web page when they click on a link to another website, or even to another of my web pages. Having people abandon my pages this way (probably) sends the wrong signals to Google and other search engines. It says I don't have relevant content. So, it's way better for SEO to open every link in a new tab. 

I wrote somewhere that I suspect you guys don't actually have experience writing posts. You are very good at what you do, but you have gone off on a tangent on this one. I think there is a term for it, tried googling but cant find it, where you are so convinced that you are right that you keep going in the wrong direction. 

As @elgato91 said, the previous settings were perfect. Maybe it's me and him alone. Just to illustrate how I use WordPress; I write maybe 5 blog articles per day. Each post may have about 4 internal links and 4 external links. I want all of them to open in a new tab. At face value, this thing may appear to be minor, but for a user, it's really, really bothersome. It bothered me so much that I spend some time researching how to give feedback to WordPress. I landed on a feedback page on WordPress.org, before being directed here by the responder. 

So far today I have written two articles. Each time that I have to add a new link, there is just this sheer annoyingness of having to click so many times. It's the reason why people are ranting. It's not like it's heavy work, but for some unknown reason, there is a sheer level of annoyingness  that comes from having to click 6 or 7 times where previously there were only 6 clicks. The first time I missed the Save button which is hidden somewhere at the bottom. Had to redo a couple of posts. 

The ideal situation would be to go back to how things were but you guys are determined to clean up the UI. I have read previous comments from July during the testing phase and a number of people raised this issue. But you guys were just determined to clean up the UI. Time could be dedicated to making tables functional, eg you cant select a row or column, and you cant use tab to move from one cell to the other. But no!!! The UI needs to be cleaned up first. Okay. Sorry. Let me head over to my yoga mat.

getdave committed:
Fix bug only allowing opens in new tab in advanced settings

getdave committed:
Augment test to assert that checking box will apply changes

getdave committed:
Implement spacing

See https://github.com/WordPress/gutenberg/pull/53566#issuecomment-1674877353

getdave commented:
@richtabor [Spacing implemented](https://github.com/WordPress/gutenberg/pull/53566#issuecomment-1674877353) and tested updated.

richtabor commented:
> it's a stepping stone whilst we figure out this positioning problem.

Good stuff. Yea I think this is a solid step in the right direction.

richtabor commented:
>  It appears you guys are concerned about aesthetics at the expense of functionality. As an example, does anyone really need that link preview? It's not necessary (in my view). Also, I don't know what the thinking was behind this entire trajectory.

Link previewing has been around for years (https://github.com/WordPress/gutenberg/issues/31466). However, I do think it's worth exploring additional editor preferences pertaining to link control, such as having a preview or not perhaps. Some folks do fine this validation affordance helpful, while others like yourself may not. 

> The ideal situation would be to go back to how things were but you guys are determined to clean up the UI. 

This is all iterative progress towards a holistic and consistent link editing experience throughout WordPress. 

In previous versions of link editing, you search/type, find your link—but don’t click on it— press the open in new tab control, then navigate back up in the UI, to finally select the link. It's not an ideal experience to navigate. 

This PR is an effort to meet in the middle; adding the control to the preview state, which renders _after_ you've identified a link. The next follow-up would be to resolve the positioning bit, which would allow the preview state to persist when selecting a link from the search results — just like it does when you paste a link onto text (https://github.com/WordPress/gutenberg/issues/53656). 

We are listening. And we all have stake in pushing the editing experience forward. I appreciate your feedback.

richtabor commented:
> I want all of them to open in a new tab. 

@TapiwaZvaks I am genuinely curious on the reasoning behind this sentiment.

richtabor reviewed (approved):
I think this is an improvement on the link editing flow. We can follow-up with design enhancements to reduce the footprint of the preview state in https://github.com/WordPress/gutenberg/issues/53654. 


| Before  | After |
| ------------- | ------------- |
|<img width="363" alt="CleanShot 2023-08-15 at 09 42 36" src="https://github.com/WordPress/gutenberg/assets/1813435/045d56f7-c63d-4642-90ac-752a63a4759a">|<img width="376" alt="CleanShot 2023-08-15 at 09 40 42" src="https://github.com/WordPress/gutenberg/assets/1813435/4867ad1a-03df-4701-b110-1bf08fd38d9c">|

TapiwaZvaks commented:
> > I want all of them to open in a new tab.
> 
> @TapiwaZvaks I am genuinely curious on the reasoning behind this sentiment.

Hi. Thanks;, here is the thinking; As a blogger, there are some on-page SEO "things," that I need to do to get my web pages noticed by Google and other search engines. Note that Google works in mysterious ways, so this aspect lies in the realm of conjecture. One (supposed) signal of how good a web page is for search engines is the amount of time that visitors spent on that page. Therein lies the usefulness of opening a post in a new tab. 

When writing a post, I usually include a link to another website in the first line. Let's say I am talking about company x. I then have a link to their home page in the first or second sentence. What would happen if someone clicks on that link and the homepage opens on top of my own article? It means a person has stayed barely a few seconds on my own web page and people usually do not bother coming back. 

There was/is actually a metric in Google analytics showing the average time spent on each web page. The best option is to have a person directed to a new tab, which leaves my own web page open. This reasoning is why it's surprising to me that such an important element is hidden. I mentioned somewhere that there was an "i" to the top left, I don't know what its called, which I used to click to see the header structure of my web page while writing a post. That's now hidden as well and it's important for SEO. It's now within a hamburger menu and I need three clicks to get there. I don't know what those other things in there are even for...

Back to the links-in-new-tab issue, the same SEO centric logic applies when it comes to internal links. I want my web pages to stay open as long as is possible. That's also important in terms of monetization. The longer someone has a page open, the more likely it is that they are going to notice an ad or promo or some other important element. 

For me personally, the best option would actually be to default all links to open in a new tab. Maybe someone will come up with a plugin or something. And maybe it's all baseless but every blogger is trying to please Google, Bing, etc. Having people abandon my pages immediately is probably not the best way of doing this.

richtabor commented:
> One (supposed) signal of how good a web page is for search engines is the amount of time that visitors spent on that page. Therein lies the usefulness of opening a post in a new tab.

I think this is also subjective; there are arguments for and against new tab, [like this one (against) from John Mueller](https://twitter.com/johnmu/status/1678463317046599688?s=61&t=OUyYdQc0IpNLvdbpDeedLg), where John states "links are links". I tend to agree, 

> For me personally, the best option would actually be to default all links to open in a new tab. Maybe someone will come up with a plugin or something. 

Maybe, but this would an lackluster experience for visitors. If every link you clicked on opened a new tab, you would end up with tens of tabs in your browser window, and you can't progress through them any longer.

Not saying there's a case when you do want them to open in a new tab (there is—hence the control within WordPress), but it likely shouldn't be the default experience for your readers. Anyhow, always conduct the research on your own accord and do what's best for your use case. 

Again, thanks for the feedback. Hoping to get this merged, stepping closer to an ideal link editing flow for everyone. 🖤

ricjcs commented:
> For me personally, the best option would actually be to default all links to open in a new tab.

I agree with almost everything you said, but on this point I disagree.

I don't think the default should be to open links in a new tab, however that process should be made easier. In my experience, most people prefer setting to open internal links in the same tab, and external links in a new tab. This for me is also what makes the most sense, opening external links in a new tab so that the visitor does not leave our website.

When making a website/posts we must think about the user and not just search engines/robots, opening all the links in a new tab does not seem to provide the best browsing experience in most cases.

> Maybe someone will come up with a plugin or something. 

There are plugins that allow you to configure this.

elgato91 commented:
When designing a blog's user experience, it is usually preferred to ensure that external links are opened in a new page, so as not to lose the reader. It's also something to do with page dwell time and Google signals, as well as UX.

But it's not a standard. Especially for internal links, if you benchmark a few hundred high-traffic WP sites you'll see that a large part decides to open it in a new tab or not, depending on the position of the link on the page. It's a matter of granular control over the user-experience. I would never force all links to open in a new tab, but I would let the creator decide what they expect their readers to do.

However, the vast majority of users already use these features thanks to third-party plugins, because WP did not have them natively. At an SEO level, however, the most important issue is the nofollow tag, also because all professional blogs heavily rely on advertising for revenue and need to adhere to specific requirements when marking affiliate links.

> I think this is also subjective; there are arguments for and against new tab, like this one (against) from John Mueller, where John states "links are links".

That quote is a bit out of context. You guys ask for feedback, but as we don't question your great work as developers, you shouldn't question ours. What we question, in fact, is your lack of sensitivity and knowledge of your user and how he uses the product. You keep answering that "it's subjective". Only your reasoning is subjective here, which goes against consolidated best practices of UX and SEO recognized practically everywhere.

Have you noticed that everyone proposing and discussing features here is a developer? Users typically don't visit GitHub. However, I haven't seen any contributions from UX/Interaction Design experts in the discussion. It appears as though you're not genuinely concerned about how WP is actually used by users, and unfortunately, the results reflect this and are far from positive.

You're making assumptions but not validating them in any way. As a result, you're convincing yourself that features like a preview are useful. However, in reality, no one seems to care about it, as it holds no value for those working with WP. It likely just adds unnecessary strain to the code and server. As others have mentioned, your efforts could have been better directed elsewhere. Would you like some validation? If, since 2021, no one has been struggling to have a preview feature, there must be a reason. If within just two days, people are persistently suggesting that you reconsider a seemingly pointless function you released, there must be a rationale behind it.

TapiwaZvaks commented:
> > For me personally, the best option would actually be to default all links to open in a new tab.
> 
> I agree with almost everything you said, but on this point I disagree.
> 
> I don't think the default should be to open links in a new tab, however that process should be made easier. In my experience, most people prefer setting to open internal links in the same tab, and external links in a new tab. This for me is also what makes the most sense, opening external links in a new tab so that the visitor does not leave our website.
> 
> When making a website/posts we must think about the user and not just search engines/robots, opening all the links in a new tab does not seem to provide the best browsing experience in most cases.
> 
> > Maybe someone will come up with a plugin or something.
> 
> There are plugins that allow you to configure this.

True enough. I have seen a lot of people stating this very same point, that external links should be opened in a new tab while internal links should be opened on the same page. I do disagree, however, that opening a link in a new tab necessarily makes for a bad user experience. I believe the opposite. If I come across a link while reading a post, and if I click on it, it doesn't mean I want to close the page that I was reading. It just means I want to reference some more information related to the subject, and I can do it in another tab.

TapiwaZvaks commented:
> > One (supposed) signal of how good a web page is for search engines is the amount of time that visitors spent on that page. Therein lies the usefulness of opening a post in a new tab.
> 
> I think this is also subjective; there are arguments for and against new tab, [like this one (against) from John Mueller](https://twitter.com/johnmu/status/1678463317046599688?s=61&t=OUyYdQc0IpNLvdbpDeedLg), where John states "links are links". I tend to agree,
> 
> > For me personally, the best option would actually be to default all links to open in a new tab. Maybe someone will come up with a plugin or something.
> 
> Maybe, but this would an lackluster experience for visitors. If every link you clicked on opened a new tab, you would end up with tens of tabs in your browser window, and you can't progress through them any longer.
> 
> Not saying there's a case when you do want them to open in a new tab (there is—hence the control within WordPress), but it likely shouldn't be the default experience for your readers. Anyhow, always conduct the research on your own accord and do what's best for your use case.
> 
> Again, thanks for the feedback. Hoping to get this merged, stepping closer to an ideal link editing flow for everyone. 🖤

Oh yes, John Mueller. He says one thing today and another tomorrow. In any case, you are right. I did mention that my response was mostly conjecture, but it was reasonable conjecture. Otherwise why would they have time spend on a page as a metric in Analytics? Why have bounce rate? If someone leaves immediately after clicking on an external link, they are essentially bouncing. In any case, don't know if those metrics are still there in GA4 but they were there in UA. 

_Maybe, but this would an lackluster experience for visitors. If every link you clicked on opened a new tab, you would end up with tens of tabs in your browser window, and you can't progress through them any longer._

I don't agree on this one. I believe opening in a new page is the most intuitive and usefully logical progression. If I am reading a how to article, do I want to have it closed because I have clicked on a link that's referenced in the article? 

I agree it shouldn't be default, but it would be great if there was an option somewhere as an example where I could be given three options 1)open all external links in a new tab or 2) open all internal links in a new tab. I could then choose to either select both or 1. Or none.

ricjcs commented:
> I believe opening in a new page is the most intuitive and usefully logical progression. If I am reading a how to article, do I want to have it closed because I have clicked on a link that's referenced in the article?

I understand your point of view and I've thought that way too. However, the website visitor can always choose to open a link in a new tab if that is their wish, so in my opinion it is more useful to force only external links to open in a new tab so that the person doesn't leave our website.

But if it's better one way or another, for this case I even think it's a little irrelevant. What matters is that the website creator can easily choose how he wants to define the links without having to do a lot of clicking.

This PR solves the problem that came up with the new WP. I'm happy about it.

> I agree it shouldn't be default, but it would be great if there was an option somewhere as an example where I could be given three options 1)open all external links in a new tab or 2) open all internal links in a new tab. I could then choose to either select both or 1. Or none.

A few years ago I opened a ticket with this suggestion on "Trac". For reasons related to accessibility it was closed.

TapiwaZvaks commented:
We are in agreement. The choice should be with the writer. And it needs to be an easy one. In any case, this has been an interesting interaction. Fruitful, at least for me. Yes, this link-in-new-tab issue will mostly be solved, but what about the "this is a sponsored link" option?  All these things were easily accessible before the latest WordPress. I don't use it personally but @elgato91 thinks its in common use.  They are all being replaced by this link preview. The link preview has no functional use. As mentioned, I spend my days typing posts. Each time I have to click six times I am drawn here and I have to chant in my head while typing responses "this is an open source project, stay polite, these guys are doing you a favor remain polite, these guys...." Lol. I am only half kidding

elgato91 commented:
@TapiwaZvaks I would like to provide a clearer explanation of my consideration on nofollow, for greater clarity in any future considerations.

Using "nofollow" to mark affiliate links is just one of the reasons (albeit a common one among professional bloggers) for using the nofollow tag. Actually, it should be used whenever you are linking to a source whose reputation you are not entirely certain about. This is because it offers a "safer" approach from an SEO perspective. Although today there are more granular controls not natively present in WP but easily addable (as those familiar with prevalent SEO plugins are aware, such as "rel=sponsored"), just a judicious use of "nofollow" suffices, according to the most recent Google guidelines.

Returning to the core issue, these controls are commonly added by major SEO plugins, so I wouldn't fret if they aren't in core. The real concern lies in the user interface's dynamics, which have shifted and, in doing so, have deviated from their main and original purpose. Consequently, it demands more effort to utilize these controls.

ricjcs commented:
I consider an option "nofollow" important and should be implemented. It's best not to depend on plugins for basic things. But in this case, although important, it doesn't seem to me to be massively used, so I wouldn't be shocked if it stayed in the "Advanced" drawer.

elgato91 commented:
@ricjcs Try checking the code of some serious site, you might discover that "nofollow" is utilized more frequently than you realize... :)

The non-use only concerns very amateur users and those who do not respect the SEO guidelines. From newspapers to professional blogs, nofollow usage is massive. In terms of importance and impact on professional user experience, I would consider these two controls to be equally important, carrying the same weight.

ellatrix commented:
This restores a change made in 6.3, right? Should we consider backporting this to a minor release? I'll add the label so it receives the attention to be discussed (not a confirmation that it will be backported).

EW0JY commented:
Hi @getdave ,

I wanted to let you know about some major issues and bugs that I've noticed with this feature.

I've updated the Gutenberg plugin to 16.5 just now to test the new link preview feature and recorded this screen capture for you:

https://github.com/WordPress/gutenberg/assets/32871574/31ff1f27-33d7-42ae-b5dc-1f987d5e1f66

As you can see, it is not working as expected at all and has the following issues:

1. the checkbox is non-responsive when clicking it.
2. worse, after clicking away from the preview, each already existing link gets actually removed.
3. after reinserting and finally managing to get the link to stay and the checkbox to be ticked, it's impossible to un-check it.

I've tested this many times and across different browsers and it's always the same.

I am not sure whether you've already been made aware of these issues and bugs. I just wanted to take the time to provide feedback anyway.

Also, I'm not familiar with Github and so I don't know if this is the correct place to provide this feedback - apologies if it isn't.
In that case, feel free to move it to the correct place.

Thank you!

getdave commented:
@EW0JY I have tested again with a fresh environment using WP 6.3 and Gutenberg 16.5.0 and I cannot replicate that error.

EW0JY commented:
@getdave Thanks for your quick reply!

Well, I have no clue as to why it doesn't happen for you.

As you can see in that video, the error is definitely occurring for me.

I've tested it in incognito in Firefox as well as Chrome.

getdave commented:
@EW0JY To help contributors I would recommend:

- Posting details of your testing environment including Operating System, Browser, WP version, Gutenberg Plugin version.
- Ensuring that you are running a clean install without Plugins and using a default Theme.

If the bug still manifests then we'll be a step closer.

I appreciate that's additional overhead but it's really the only way to narrow this down.

If you can still replicate then it will also help if this is posted as a [new Issue](https://github.com/WordPress/gutenberg/issues/new/choose) as that will cause it to be picked up by Triage teams and thus get more testers involved 🙏 

Thanks for your help here.

TapiwaZvaks commented:
I was able to "almost" replicate the issue. Things work but they are a bit iffy. When I insert a link using ctrl + v, I am able to click on the checkbox and everything works fine. However, when I manually type in a link as @EW0JY did in the screenshot, things become a bit patchy. Clicking on the checkbox does nothing. However, when I click on the side and then click on the link one more time, I can see that the box is already checked.

EW0JY commented:
@getdave I understand, so I went through all the steps you recommended and did the following:

I installed a fresh WP version without any plugins and could confirm that the link feature works as expected.

So I cloned my production site and deactivated all plugins, after which the problem also disappeared.

Then I reactivated each plugin one by one until the problem reoccurred.

I can now definitely confirm that the problem is this:

The new link feature is incompatible with the YOAST SEO plugin.

That is the only plugin I have to deactivate to make the feature work as expected.

I also downgraded to a previous version of the yoast plugin, but even then it didn't work.

Since it's such a popular plugin and not feasible for me to stop using it, I hope that is not what you are going to suggest as a "solution".

Also, due to its popularity, a ton of people will also be affected by it.

Can you please install the yoast plugin, replicate the issue and make the link feature compatible with yoast?

Oh, my environment is Windows 10 home, latest Firefox and Chrome versions, WP 6.3 with the latest Twenty-Twentytwo theme and Gutenberg Plugin 16.5.

I hope that helps, thank you!

P.S.:

I've tested a bit more and was able to narrow it down to this toggle within the yoast plugin settings:

![conflicting yoast setting](https://github.com/WordPress/gutenberg/assets/32871574/f32bb84c-ee6f-44be-820e-c6bcaf910f63)

When I disable that toggle, your link preview feature works as expected.

In my specific case, I think it's a viable workaround as I don't need those "additional settings" enabled - for other people, this may be different.

I hope this helps,
Best wishes!

ricjcs commented:
I was able to reproduce the same issue. Without the Yoast plugin it works perfectly, with Yoast activated the reported problem happens, I also tested it with Rank Math, the same thing happens.

draganescu commented:
#53949 is fixing the bug described by @EW0JY and reproduced by @ricjcs . Thank you both for the great steps.

EW0JY commented:
@draganescu I can't seem to find #81960 when searching for it on Github - could you please link to it?

Thanks!

draganescu commented:
It's #53949 @EW0JY 🤕
